### PR TITLE
feat: get/set global filters state with state API

### LINKED ISSUE #505: Default filters can be modified via `setState` API
Use case:
The user is linked to an AppMap with a selected event that is not visible under the default filters. For example, a query that occurs between HTTP web requests can not be selected due to the default filter to restrict root events to HTTP.

### PR DESCRIPTION
**Pull request by sashahonchar**
Fixes #505

### CODE REVIEW

**Review by dustinbyrne — CHANGES_REQUESTED** [review]
The change looks good but this needs test cases.

**Comment by sashahonchar** [comment]
@dustinbyrne  Found a bug in code and added unit test. https://github.com/applandinc/appmap-js/pull/509/commits/b5a1835fa8ac26ccaad3d22f7dfa106e45aeff92

File: packages/components/tests/unit/VsCodeExtension.spec.js [hunk 1]
```diff
@@ -27,6 +27,16 @@ describe('VsCodeExtension.vue', () => {
...
+
+    wrapper.vm.setState(
+      '{"currentView":"viewFlow","filters":{"rootObjects":[],"limitRootEvents":false,"hideMediaRequests":true,"hideUnlabeled":false,"hideElapsedTimeUnder":100,"hideName":false}}'
+    );
+    expect(wrapper.vm.filters.declutter.limitRootEvents.on).toBe(false);
+    expect(wrapper.vm.filters.declutter.hideMediaRequests.on).toBe(true);
+    expect(wrapper.vm.filters.declutter.hideUnlabeled.on).toBe(false);
+    expect(wrapper.vm.filters.declutter.hideElapsedTimeUnder.on).toBe(true);
+    expect(wrapper.vm.filters.declutter.hideElapsedTimeUnder.time).toBe(100);
+    expect(wrapper.vm.filters.declutter.hideName.on).toBe(false);
```
Comment (dustinbyrne) [inline packages/components/tests/unit/VsCodeExtension.spec.js:39]: `limitRootEvents` / `hideUnlabeled` / `hideName` properties have no test coverage in this case.

**Review by dustinbyrne — CHANGES_REQUESTED** [review]
Looking better. We'll also need an updated test for `getState`, where the filters contain non-default values. Plus a comment below.

**Review by dustinbyrne — APPROVED** [review]

**Pull request merged**